src/utils/cpUtils.ts

```diff
@@ -9,60 +9,92 @@ import * as vscode from 'vscode';
+                        this.outputChannel.append(data);
                     }
-                } else {
-                    if (outputChannel) {
-                        outputChannel.appendLine(localize('finishedRunningCommand', 'Finished running command: "{0} {1}".', command, formattedArgs));
+                });
+
+                childProc.stderr.on('data', (data: string | Buffer) => {
+                    data = data.toString();
+                    this.cmdOutputIncludingStderr = this.cmdOutputIncludingStderr.concat(data);
+                    if (this.outputChannel) {
+                        this.outputChannel.append(data);
                     }
-                    resolve();
-                }
+                });
+
+                childProc.on('error', (err: Error) => {
+                    this.onErrorCallback(err, reject);
+                });
+
+                childProc.on('close', (code: number) => {
+                    this.onCloseCallback(code, resolve, reject);
+                });
             });
-        });
 
-        return cmdOutput;
+            return this.cmdOutput;
+        }
+
+        protected onCloseCallback(code: number, resolve: () => void, reject: (e: Error) => void): void {
+            if (code !== 0) {
+                // We want to make sure the full error message is displayed to the user, not just the error code.
+                // If outputChannel is defined, then we simply call 'outputChannel.show()' and throw a generic error telling the user to check the output window
+                // If outputChannel is _not_ defined, then we include the command's output in the error itself and rely on AzureActionHandler to display it properly
+                if (this.outputChannel) {
+                    this.outputChannel.show();
+                    reject(new Error(localize('azFunc.commandErrorWithOutput', 'Failed to run "{0}" command. Check output window for more details.', this.command)));
+                } else {
+                    reject(new Error(localize('azFunc.commandError', 'Command "{0} {1}" failed with exit code "{2}":{3}{4}', this.command, this.formattedArgs, code, os.EOL, this.cmdOutputIncludingStderr)));
+                }
+            } else {
+                if (this.outputChannel) {
+                    this.outputChannel.appendLine(localize('finishedRunningCommand', 'Finished running command: "{0} {1}".', this.command, this.formattedArgs));
+                }
+                resolve();
+            }
+        }
+
+        protected onErrorCallback(err: Error, reject: (e: Error) => void): void {
+            reject(err);
+        }
+    }
+
+    export async function executeCommand(outputChannel: vscode.OutputChannel | undefined, workingDirectory: string | undefined, command: string, ...args: string[]): Promise<string> {
+        const childProcessExecutor: ChildProcessExecutor = new ChildProcessExecutor(outputChannel, workingDirectory, command, args);
```

Comment:
I'm mostly okay with _what_ you did, but not quite _how_ you did it. Specifically, I'm not a fan of creating a class, immediately using it once, and never using it again. To me, that implies it should be a function instead. How about you structure the code more like this:
```typescript
export namespace cpUtils {
    export async function executeCommand(...): Promise<string> {
        const result: ICommandResult = await tryExecuteCommand(...);
        // most of the stuff from ChildProcessExecutor.onCloseCallback
        return result.cmdOutput;
    }

    export async function tryExecuteCommand(...): Promise<ICommandResult> {

    }

    export interface ICommandResult {
        code: number;
        cmdOutput: string;
        cmdOutputIncludingStderr: string;
        formattedArgs: string;
    }
}

export namespace mavenUtils {
    export async function executeMvnCommand(...): Promise<string> {
        const result: ICommandResult = await tryExecuteCommand(...);
        // most of the stuff from MavenCommandExecutor.onCloseCallback
        return result.cmdOutput;
    }
}
```